lib/takeaway.rb

```diff
@@ -0,0 +1,46 @@
+class Takeaway
+  attr_reader :menu
+
+  def initialize(menu, messager = Messager)
+    @menu = menu
+    @messager = messager
+  end
+
+  def calculate_price(order)
+    order.inject(0) do |sum, item|
+      name = item[0]
+      quantity = item[1]
+      sum + menu.price_of(name) * quantity
+    end
+  end
+
+  def receive_order(order, payment)
+    fail 'Cannot process an empty order' if order.empty?
+    validate_order_items(order)
+    validate_payment(order, payment)
+    messager.send_message(order.customer.phone, create_message)
+  end
+
+  private
+
+  attr_reader :messager
+
+  def validate_order_items(order)
+    order.each do |name, quantity|
```

Comment:
Lint/UnusedBlockArgument: Unused block argument - `quantity`. If it's necessary, use `_` or `_quantity` as an argument name to indicate that it won't be used. (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)